source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -248,8 +250,14 @@ void CheckRequestUtils::createHttpCheck(
                         cb->decodingBuffer(), headers, max_request_bytes, pack_as_bytes,
                         encode_raw_headers, request_header_matchers);
 
-  if (include_tls_session && cb->connection()->ssl() != nullptr) {
-    setTLSSession(*attrs->mutable_tls_session(), cb->connection()->ssl());
+  if (include_tls_session) {
+    // Try to get the SNI from the TLS session. If not available there then try the
+    // data from the TLS inspector (i.e. the server name)
+    if (cb->connection()->ssl() != nullptr) {
```

Comment:
To avoid the copy/paste, I'd change `setTLSSession()` to take `cb->connection()` as the second parameter, and put the logic of `connection->ssl() != nullptr` in that function, and then get `connection->requestedServerName()` as the alternative in that function. Also, if `requestedServerName().empty()` is true, don't make the call to `set_sni()` because `mutable_tls_session()` would then have the side effect of creating an empty message; no message at all is the correct behavior in this case.

Reply:
I made this change. Thanks for the recommendation.